akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -1443,9 +1498,12 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
    *
    * This allows the dynamic creation of an Outlet for a GraphStage which is
    * connected to a Source that is available for materialization (e.g. using
-   * the `subFusingMaterializer`). Care needs to be taken to complete this
-   * Outlet when the operator shuts down lest the corresponding Sink be left
-   * hanging. It is good practice to use the `timeout` method to cancel this
+   * the `subFusingMaterializer`). Completion, cancellation and failure of the
+   * parent operator is automatically delegated to instances of `SubSourceOutlet`
+   * to avoid resource leaks.
+   *
+   * FIXME: is this advice still needed:
```

Comment:
Good point, it's not needed any more to avoid a leak but might still be useful to prevent that something gets stuck?

Reply:
Rephrased a tiny bit and kept recommendation.